### PR TITLE
Rand soa integration

### PR DESCRIPTION
**Pull request by kkahadze**

### CODE REVIEW

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 1]
```diff
@@ -143,6 +201,26 @@ def is_support_for_selected_orders(self, support_order: str) -> bool:
             return True
         return False
 
+    def is_support_for_given_orders(self, support_order, orders):
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:204]: please add type hints and sphinx-doc parameter descriptions

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 2]
```diff
@@ -143,6 +201,26 @@ def is_support_for_selected_orders(self, support_order: str) -> bool:
...
+        """Determine if selected support order for neighbour corresponds to given list of orders"""
+        order_tokens = get_order_tokens(support_order)
+        if order_tokens[2].split()[1] not in orders:
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:207]: add a comment explaining this

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 3]
```diff
@@ -143,6 +201,26 @@ def is_support_for_selected_orders(self, support_order: str) -> bool:
...
+            return True  # it's okay to support other power than allies
+        selected_order = get_order_tokens(orders[order_tokens[2].split()[1]])
+
+        if (
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:211]: add comments throughout.

Additionally, you may want to turn the auto formatter off in some sections

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 4]
```diff
@@ -278,12 +357,23 @@ def generate_support_proposals(self, comms_obj: MessagesData) -> Dict[str, str]:
                 final_messages[self.game._unit_owner(selected_order[0]).name].append(
                     selected_order[2]
                 )
-            
+
             for recipient in final_messages:
                 # Construct message for each support proposal
-                if not(final_messages[recipient]):
+                if not (final_messages[recipient]):
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:363]: remove (parens)

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 5]
```diff
@@ -298,21 +388,28 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
...
-        self.alliances =  parsed_messages_dict['alliance_proposals']
+        parsed_messages_dict = parse_proposal_messages(
+            rcvd_messages, self.game, self.power_name
+        )
+        valid_proposal_orders = parsed_messages_dict["valid_proposals"]
+        invalid_proposal_orders = parsed_messages_dict["invalid_proposals"]
+        shared_orders = parsed_messages_dict["shared_orders"]
+        other_orders = parsed_messages_dict["other_orders"]
+        self.alliances = parsed_messages_dict["alliance_proposals"]
+        # print("debug: Parsed proposal messages")
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:399]: delete commented out code

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 6]
```diff
@@ -324,18 +421,66 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
         msgs_data = self.gen_messages(orders_data.get_list_of_orders())
         self.respond_to_invalid_orders(invalid_proposal_orders, msgs_data)
         self.respond_to_alliance_messages(msgs_data)
-        msg_allies = ','.join([pow for pow in self.stance.stance[self.power_name] if (pow != self.power_name and self.stance.stance[self.power_name][pow] > 0)])
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:327]: add format off comments here

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 1 above)
Comment (kkahadze) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:204]: I didn't write this function so I think it is best for somebody else to add comments and type hints. I can make an issue if you'd like.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 2 above)
Comment (kkahadze) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:207]: I didn't write this function so I think it is best for somebody else to add comments and type hints. I can make an issue if you'd like.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 3 above)
Comment (kkahadze) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:211]: I didn't write this part so I think it is best for somebody else to add comments and type hints. I can make an issue if you'd like.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 4 above)
Comment (kkahadze) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:363]: I didn't write this but it was caused by the formatter I believe, is it worth it to turn off formatting with comments just to fix that?

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 6 above)
Comment (kkahadze) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:327]: Done, I also made the lists of allies, foes and neutrals separate vars for readability.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 5 above)
Comment (kkahadze) [reply src/baseline_bots/bots/smart_order_accepter_bot.py]: Done

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 1 above)
Comment (kartik2112) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:202]: I wrote the get_support_proposals related functions. I will work on doing the cosmetic changes to those functions in a separate PR.

**Review by kartik2112 — APPROVED** [review]

**Pull request merged**